crates/egui/src/widgets/drag_value.rs

```diff
@@ -170,6 +198,170 @@ impl<'a> DragValue<'a> {
         self
     }
+
+    /// Set custom parser defining how the text input is parsed into a number.
+    ///
+    /// A custom parser takes an `&str` to parse into a number and returns a `f64` if it was successfully parsed
+    /// or `None` otherwise.
+    ///
+    /// See also: [`DragValue::custom_formatter`]
+    ///
+    /// ```
+    /// # egui::__run_test_ui(|ui| {
+    /// # let mut my_i32: i32 = 0;
+    /// ui.add(egui::DragValue::new(&mut my_i32)
+    ///     .clamp_range(0..=((60 * 60 * 24) - 1))
+    ///     .custom_formatter(|n, _| {
+    ///         let n = n as i32;
+    ///         let hours = n / (60 * 60);
+    ///         let mins = (n / 60) % 60;
+    ///         let secs = n % 60;
+    ///         format!("{hours:02}:{mins:02}:{secs:02}")
+    ///     })
+    ///     .custom_parser(|s| {
+    ///         let parts: Vec<&str> = s.split(':').collect();
+    ///         if parts.len() == 3 {
+    ///             parts[0].parse::<i32>().and_then(|h| {
+    ///                 parts[1].parse::<i32>().and_then(|m| {
+    ///                     parts[2].parse::<i32>().map(|s| {
+    ///                         ((h * 60 * 60) + (m * 60) + s) as f64
+    ///                     })
+    ///                 })
+    ///             })
+    ///             .ok()
+    ///         } else {
+    ///             None
+    ///         }
+    ///     }));
+    /// # });
+    /// ```
+    pub fn custom_parser(mut self, parser: impl 'a + Fn(&str) -> Option<f64>) -> Self {
+        self.custom_parser = Some(Box::new(parser));
+        self
+    }
+
+    /// Set `custom_formatter` and `custom_parser` to display and parse numbers in binary.
+    /// Signed integers and floating point numbers are not supported.
+    ///
+    /// `min_width` specifies the minimum number of displayed digits; if the number is shorter than this, it will be
+    /// prefixed with additional 0s to match `min_width`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `min_width` is 0.
+    ///
+    /// ```
+    /// # egui::__run_test_ui(|ui| {
+    /// # let mut my_i32: i32 = 0;
+    /// ui.add(egui::DragValue::new(&mut my_i32).binary_u64(64));
+    /// # });
+    /// ```
+    pub fn binary_u64(self, min_width: usize) -> Self {
+        assert!(
+            min_width > 0,
+            "DragValue::binary_u64: `min_width` must be greater than 0"
+        );
+        self.custom_formatter(move |n, _| format!("{:0>min_width$b}", n as u64))
+            .custom_parser(|s| {
+                if s.is_empty() {
+                    return None;
+                }
+                let num_digits = s.len().min(64);
+                let mut n = 0;
+                for (i, digit) in s.chars().rev().take(num_digits).enumerate() {
+                    if digit != '0' && digit != '1' {
+                        return None;
+                    }
+                    n |= (digit as u64 - '0' as u64) << (i as u64);
+                }
```

Comment:
Use https://doc.rust-lang.org/stable/std/primitive.i64.html#method.from_str_radix here and below - less code, and automatic support for signed integers!

Reply:
I was thinking about using that but wasn't sure if we wanted another dependency. I'll add it now!

Reply:
Oh cool, I don't know how I missed this!

Reply:
@emilk According to [`Binary`]() documentation:

> For primitive signed integers ([i8](https://doc.rust-lang.org/stable/std/primitive.i8.html) to [i128](https://doc.rust-lang.org/stable/std/primitive.i128.html), and [isize](https://doc.rust-lang.org/stable/std/primitive.isize.html)), negative values are formatted as the two’s complement representation.

Are we happy with this?

Reply:
I added a bool that lets you choose if you want the number to be displayed with a sign or in 2's complement.